audb/core/dependencies.py

```diff
@@ -375,52 +372,79 @@ def _add_media(
 
         Args:
             root: root directory
-            file: relative file path
-            archive: archive name without extension
-            checksum: checksum of file
-            version: version string
+            media: list of tuples with (
+                    relative file path,
+                    version string,
+                    archive name without extension,
+                    checksum,
+                )
 
         """
-        format = audeer.file_extension(file).lower()
+        if not media:
+            return
+
+        files = []
+        rows = []
+
+        for file, version, archive, checksum in media:
```

Comment:
We could also use threading here. But let first see what we gained with the changes so far.

Reply:
I did a first test locally using `num_workers=8` and added the needed code here. I tested with adding 1000 new files to `emodb` there it did not really help, but was slightly slower. But of cause it could still be that for a very large number of files it will be faster. I would also vote for ignoring this in this pull request.

Reply:
What I have seen in the past is that when you request the durations of a large number of files for the first time it can be quite slow, but then when you re-run the same code again it's suddenly very fast. So it seems there is some caching involved by the file system. Maybe restart your machine before you try with multiple thread if you have tried with single thread already?

Reply:
Created https://github.com/audeering/audb/issues/142 to further track this.